src/lua/init.lua

```diff
@@ -86,81 +111,85 @@ local function load_lua(file)
     return loadfile(file)
 end
 
-local function search_cwd_lib(name)
-    local path = "./?."..soext
-    return package.searchpath(name, path)
-end
+local function merge_strings(paths, templates)
+    local searchpaths = {}
+    paths = type(paths) == 'table' and paths or { paths }
+    templates = templates or {}
 
-local function search_cwd_lua(name)
-    local path = "./?.lua;./?/init.lua"
-    return package.searchpath(name, path)
-end
+    if #templates == 0 then
+        return paths
+    end
 
-local function traverse_rocks(name, pathes_search)
-    local cwd = fio.cwd()
-    local index = string.len(cwd) + 1
-    local strerr = ""
-    while index ~= nil do
-        cwd = string.sub(cwd, 1, index - 1)
-        for i, path in ipairs(pathes_search) do
-            local file, err = package.searchpath(name, cwd .. path)
-            if err == nil then
-                return file
-            end
-            strerr = strerr .. err
+    if #paths == 0 then
+        return templates
+    end
+
+    for _, path in ipairs(paths) do
+        for _, template in pairs(templates) do
+            table.insert(searchpaths, fio.pathjoin(path, template))
         end
-        index = string.find(cwd, "/[^/]*$")
     end
-    return nil, strerr
-end
 
-local function search_rocks_lua(name)
-    local pathes_search = {
-        "/.rocks/share/tarantool/?.lua;",
-        "/.rocks/share/tarantool/?/init.lua;",
-    }
-    return traverse_rocks(name, pathes_search)
+    return searchpaths
 end
 
-local function search_rocks_lib(name)
-    local pathes_search = {
-        "/.rocks/lib/tarantool/?."..soext
-    }
-    return traverse_rocks(name, pathes_search)
+local function traverse_path(path)
+    path = path:startswith('/') and path or fio.abspath(path)
+    local paths = { path }
+
+    while path ~= '/' do
+        path = fio.dirname(path)
+        table.insert(paths, path)
+    end
+
+    return paths
 end
 
-local function cwd_loader_func(lib)
-    local search_cwd = lib and search_cwd_lib or search_cwd_lua
-    local load_func = lib and load_lib or load_lua
+local function gen_search_func(opts)
```

Comment:
Can you add an API comment here and for gen_loader_func()?

Reply:
Added comments for both functions

Reply:
Done